services/app-api/src/postgres/contractAndRates/withdrawContract.ts

```diff
@@ -237,18 +291,141 @@ const withdrawContractInsideTransaction = async (
             : ''
 
     // unlock contract with withdraw default message and withdraw input reason
-    await unlockContractInsideTransaction(tx, {
+    const unlockContract = await unlockContractInsideTransaction(tx, {
         contractID: contract.id,
         unlockedByUserID: updatedByID,
         unlockReason: `CMS withdrawing the submission ${contractName}${rateNamesMessage}. ${updatedReason}`,
     })
 
-    // resubmit contract with the default message and withdraw input reason
-    await submitContractInsideTransaction(tx, {
-        contractID: contract.id,
-        submittedByUserID: updatedByID,
-        submittedReason: `CMS has withdrawn the submission ${contractName}${rateNamesMessage}. ${updatedReason}`,
-    })
+    if (unlockContract instanceof Error) {
+        throw new Error(
+            `Cannot unlock contract to be withdrawn. ${unlockContract.message}`
+        )
+    }
+
+    //NOTE: Consider a separate function to do the reassignment
```

Comment:
issue: Please break this code up now before merging, make the ` reassignRateParentContractInTransaction`.